File: frontend/src/types/project.ts
import type { ProjectStatus, ProjectType } from './enums'

export interface Project {
  id: string
  owner_id: string
  name: string
  description: string
  country: string
  region: string
  project_type: ProjectType
  status: ProjectStatus
  start_date: string
  end_date: string | null
  color: string
  created_at: string
  updated_at: string
  // Computed, server-derived fields present on every representation.
  site_count: number
  total_area_hectares: number
  total_carbon_tco2e: number
  avg_biodiversity_score: number | null
}

export interface ProjectListResponse {
  items: Project[]
  total: number
}

export interface ProjectUpsertRequest {
  name: string
  description: string
  country: string
  region: string
  project_type: ProjectType
  status: ProjectStatus
  start_date: string
  end_date: string | null
  color: string
}

export interface ProjectListParams {
  search?: string
  status?: ProjectStatus
}

File: backend/app/schemas/project.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel, Field

from app.models.enums import ProjectStatus, ProjectType


class ProjectCreate(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    description: str = ""
    country: str = Field(min_length=1, max_length=120)
    region: str = Field(min_length=1, max_length=120)
    project_type: ProjectType
    status: ProjectStatus
    start_date: date
    end_date: date | None = None
    color: str = Field(default="#1b4332", pattern=r"^#[0-9A-Fa-f]{6}$")


class ProjectUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=255)
    description: str | None = None
    country: str | None = Field(default=None, min_length=1, max_length=120)
    region: str | None = Field(default=None, min_length=1, max_length=120)
    project_type: ProjectType | None = None
    status: ProjectStatus | None = None
    start_date: date | None = None
    end_date: date | None = None
    color: str | None = Field(default=None, pattern=r"^#[0-9A-Fa-f]{6}$")


class ProjectRead(BaseModel):
    id: uuid.UUID
    owner_id: uuid.UUID
    name: str
    description: str
    country: str
    region: str
    project_type: ProjectType
    status: ProjectStatus
    start_date: date
    end_date: date | None
    color: str
    created_at: datetime
    updated_at: datetime
    site_count: int
    total_area_hectares: float
    total_carbon_tco2e: float
    avg_biodiversity_score: float | None


class ProjectListResponse(BaseModel):
    items: list[ProjectRead]
    total: int
